security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -244,32 +262,44 @@ func (sf *SecretFetcher) scrtUpdated(oldObj, newObj interface{}) {
 
 	t := time.Now()
 	ns := &model.SecretItem{
-		ResourceName:     newScrtName,
-		CertificateChain: newCert,
-		PrivateKey:       newKey,
-		CreatedTime:      t,
-		Version:          t.String(),
+		ResourceName: newScrtName,
+		CreatedTime:  t,
+		Version:      t.String(),
 	}
+
+	rootNotice := ""
+	if isK8SCACert {
+		ns.RootCert = newRoot
+		ns.RootCertOwnedByCompoundSecret = false
+		rootNotice = " as a client root CA"
+	} else {
+		ns.CertificateChain = newCert
+		ns.PrivateKey = newKey
+	}
+
 	sf.secrets.Store(newScrtName, *ns)
-	log.Debugf("secret %s is updated", newScrtName)
+	log.Debugf("secret %s is updated%s", newScrtName, rootNotice)
 	if sf.UpdateCache != nil {
 		sf.UpdateCache(newScrtName, *ns)
 	}
 
-	rootCertResourceName := newScrtName + IngressGatewaySdsCaSuffix
-	// If there is root cert secret with the same resource name, delete that secret now.
-	sf.secrets.Delete(rootCertResourceName)
-	if len(newRoot) > 0 {
-		nsRoot := &model.SecretItem{
-			ResourceName: rootCertResourceName,
-			RootCert:     newRoot,
-			CreatedTime:  t,
-			Version:      t.String(),
-		}
-		sf.secrets.Store(rootCertResourceName, *nsRoot)
-		log.Debugf("secret %s is updated", rootCertResourceName)
-		if sf.UpdateCache != nil {
-			sf.UpdateCache(rootCertResourceName, *nsRoot)
+	if !isK8SCACert && len(newRoot) > 0 {
```

Comment:
Shall we change it to:
if !isK8SCACert {
  rootCertResourceName := newScrtName + IngressGatewaySdsCaSuffix
  if len(oldRoot) > 0 {
    sf.secrets.Delete(rootCertResourceName)
    if len(newRoot) == 0 && sf.DeleteCache != nil {
	sf.DeleteCache(rootCertResourceName)      
    }
  }
  if len(newRoot) > 0 {
  ......
  }
}
When newRoot is empty, but oldRoot is not empty, we don't want to use the oldRoot.

Reply:
Done.